routers/common/markup.go

```diff
@@ -66,6 +66,23 @@ func RenderMarkup(ctx *context.Base, repo *context.Repository, mode, text, urlPr
 		}
 	}
 
+	links := markup.Links{
+		AbsolutePrefix: true,
+		Base:           urlPrefix,
+	}
+
+	// Parse branch path and tree path, for correct media links.
+	// The expected route is "{user}/{repo}/src/{branch, commit, tag]/{identifier}"
+	urlElements := strings.Split(strings.TrimPrefix(urlPrefix, setting.AppURL), "/")
+	if len(urlElements) >= 5 && urlElements[2] == "src" {
+		links = markup.Links{
+			AbsolutePrefix: true,
+			Base:           setting.AppURL + urlElements[0] + "/" + urlElements[1],
+			BranchPath:     urlElements[3] + "/" + urlElements[4],
```

Comment:
Is it right for all cases? For example, a branch named `fix/bug-1`, then the URL is `https://.../fvcorp-dev/fvweb2/src/branch/fix/bug-1/.....`, then `urlElements[3] + "/" + urlElements[4]` is not the right path.

Reply:
Indeed, that's an issue. In the current code `Links` only ever concatenates `BranchPath` and `TreePath` so it works ok. But it could be a problem in the future.

If this needs to be solved, potential solutions:
* Change `markup.Links` to only store a single path instead of separate `BranchPath` and `TreePath`.
* Refactor `context.RepoRefByType` and related functions as used in routing so they can be used with an arbitrary `urlPrefix` string instead of assuming `ctx.Params`.

Reply:
@KN4CK3R do you have some ideas? I think the new design could depend on the the plan of Rework markup link rendering (#26745)